docs/conceptual/core/codec.md

```diff
@@ -1,13 +1,146 @@
+
+- `0b00`: single-byte mode; upper six bits are the LE encoding of the value (valid only for values of 0-63).
+- `0b01`: two-byte mode: upper six bits and the following byte is the LE encoding of the value (valid only for values `64-(2**14-1)`).
+- `0b10`: four-byte mode: upper six bits and the following three bytes are the LE encoding of the value (valid only for values `(2**14-1)-(2**30-1)`).
+- `0b11`: Big-integer mode: The upper six bits are the number of bytes following, less four. The value is contained, LE encoded, in the bytes following. The final (most significant) byte must be non-zero. Valid only for values `(2**30-1)-(2**536-1)`.
+
+#### Example
+
+- `unsigned integer 0`: `0x00`
+- `unsigned integer 1`: `0x04`
+- `unsigned integer 42`: `0xa8`
+- `unsigned integer 69`: `0x1501`
+
+Error:
+
+- ~~`0x0100`: Zero encoded in mode 1~~
+
+### Options
+
+One or zero values of a particular type. Encoded as:
+
+- `0x00` if it is `None` ("empty" or "null").
+- `0x01` followed by the encoded value if it is `Some`.
+
+As an exception, in the case that the type is a boolean, then it is always one byte:
+
+- `0x00` if it is `None` ("empty" or "null").
+- `0x01` if it is the `false` value.
+- `0x02` if it is the `true` value.
+
+### Vectors (lists, series, sets)
+
+A collection of same-typed values is encoded, prefixed with a *compact* encoding of the number of items, followed by each item's encoding concatenated in turn.
+
+#### Example
+
+Vector of unsigned 16-bit integers:
+
+```
+[4, 8, 15, 16, 23, 42]
+```
+
+SCALE Bytes:
+
+```
+0x18040008000f00100017002a00
+```
+
+### Tuples and Structures
+
+A fixed-size series of values, each with a possibly different but predetermined and fixed type. This is simply the concatenation of each encoded value. For structures, the values are named, but that is irrelevant for the encoding (names are ignored - only order matters).
+
+#### Example
+
+Tuple of compact unsigned integer and boolean:
+
+`(3, false)`: `0x0c00`
+
+### Enumerations (tagged-unions)
+
+A fixed number of variants, each mutually exclusive and potentially implying a further value or series of values.
+
+Encoded as the first byte identifying the index of the variant that the value is. Any further bytes are used to encode any data that the variant implies.
+
+#### Example
+
+```rust
+enum IntOrBool {
+  Int(u8),
+  Bool(bool),
+}
+```
+
+- `Int(42)`: `0x002a`
+- `Bool(true)`: `0x0101`
+
+## Implementations
+
+The Parity SCALE Codec is currently implemented for:
+
+* Rust: [`paritytech/parity-scale-codec`](https://github.com/paritytech/parity-scale-codec)
+* Python: [`polkascan/py-scale-codec`](https://github.com/polkascan/py-scale-codec)
+* Golang: [`ChainSafe/gossamer`](https://github.com/ChainSafe/gossamer)
+* C++: [`soramitsu/scale`](https://github.com/soramitsu/scale)
+* JavaScript: [`polkadot-js/api`](https://github.com/polkadot-js/api)
+
+## Next Steps
+
+### Learn More
+
+TODO
+
+### Examples
+
```

Comment:
I'm comfortable omitting this since you have examples throughout.

Reply:
Yeah, we can remove when we do a "final pass" on next steps at the end of writing most the docs.

Otherwise, it might be cool to just point to some of the in-code tests we have for the scale codec to see how it is pracitcally used.

Another good example might be to simply show how you can call the `.encode()` function on items to turn it into it's scale encoded format.

Reply:
Encoding and test examples are going to use a specific representation. I'm fine with that though. Would be cool to show examples in s couple different languages.

Reply:
It would be from Substrate, which makes sense in the context of Substrate docs eh?